Makefile

```diff
@@ -21,49 +21,52 @@ test: $(CPDEPS) $(DPDEPS)
 DOCKER_REGISTRY          ?=
 DOCKER_REPOSITORY        ?=
-DOCKER_BUILD_ARGS        ?=
+DOCKER_BUILD_ARGS        ?= --build-arg RTE_MACHINE='hsw'
 
 ## Docker labels. Only set ref and commit date if committed
 DOCKER_LABEL_VCS_URL     ?= $(shell git remote get-url $(shell git remote))
 DOCKER_LABEL_VCS_REF     ?= $(shell git diff-index --quiet HEAD -- && git rev-parse HEAD || echo "unknown")
 DOCKER_LABEL_COMMIT_DATE ?= $(shell git diff-index --quiet HEAD -- && git show -s --format=%cd --date=iso-strict HEAD || echo "unknown" )
 DOCKER_LABEL_BUILD_DATE  ?= $(shell date -u "+%Y-%m-%dT%H:%M:%SZ")
-
-CP_NAME                  ?= cp
-DP_NAME                  ?= dp
-CP_IMAGENAME             ?= ${DOCKER_REGISTRY}${DOCKER_REPOSITORY}ngic-${CP_NAME}:${DOCKER_TAG}
-DP_IMAGENAME             ?= ${DOCKER_REGISTRY}${DOCKER_REPOSITORY}ngic-${DP_NAME}:${DOCKER_TAG}
+DOCKER_TARGETS           ?= cp dp
 
 # https://docs.docker.com/engine/reference/commandline/build/#specifying-target-build-stage---target
 docker-build:
-	docker build $(DOCKER_BUILD_ARGS) \
-		--target ${CP_NAME} \
-		--tag ${CP_IMAGENAME} \
-		--label "org.label-schema.schema-version=1.0" \
-		--label "org.label-schema.name=ngic-${CP_NAME}" \
-		--label "org.label-schema.version=${VERSION}" \
-		--label "org.label-schema.vcs-url=${DOCKER_LABEL_VCS_URL}" \
-		--label "org.label-schema.vcs-ref=${DOCKER_LABEL_VCS_REF}" \
-		--label "org.label-schema.build-date=${DOCKER_LABEL_BUILD_DATE}" \
-		--label "org.opencord.vcs-commit-date=${DOCKER_LABEL_COMMIT_DATE}" \
-                .
-	docker build $(DOCKER_BUILD_ARGS) \
-		--target ${DP_NAME} \
-		--tag ${DP_IMAGENAME} \
-		--label "org.label-schema.schema-version=1.0" \
-		--label "org.label-schema.name=ngic-${DP_NAME}" \
-		--label "org.label-schema.version=${VERSION}" \
-		--label "org.label-schema.vcs-url=${DOCKER_LABEL_VCS_URL}" \
-		--label "org.label-schema.vcs-ref=${DOCKER_LABEL_VCS_REF}" \
-		--label "org.label-schema.build-date=${DOCKER_LABEL_BUILD_DATE}" \
-		--label "org.opencord.vcs-commit-date=${DOCKER_LABEL_COMMIT_DATE}" \
-                .
+	for target in $(DOCKER_TARGETS); do \
+		docker build $(DOCKER_BUILD_ARGS) \
+			--target $$target \
+			--tag ${DOCKER_REGISTRY}${DOCKER_REPOSITORY}ngic-$$target:${DOCKER_TAG} \
+			--label "org.label-schema.schema-version=1.0" \
+			--label "org.label-schema.name=ngic-$$target" \
+			--label "org.label-schema.version=${VERSION}" \
+			--label "org.label-schema.vcs-url=${DOCKER_LABEL_VCS_URL}" \
+			--label "org.label-schema.vcs-ref=${DOCKER_LABEL_VCS_REF}" \
+			--label "org.label-schema.build-date=${DOCKER_LABEL_BUILD_DATE}" \
+			--label "org.opencord.vcs-commit-date=${DOCKER_LABEL_COMMIT_DATE}" \
+			.; \
+		docker build $(DOCKER_BUILD_ARGS) \
+			--target $$target \
+			--tag ${DOCKER_REGISTRY}${DOCKER_REPOSITORY}ngic-$$target:${DOCKER_DEBUG_TAG} \
+			--build-arg RUN_BASE="runtime-utils" \
+			--build-arg EXTRA_CFLAGS="-DUSE_AF_PACKET -DSTATIC_ARP -UPERF_TEST -ggdb -O2" \
```

Comment:
@hyunsun can you say more about `-DSTATIC_ARP -UPERF_TEST`

Reply:
STATIC_ARP is required when main interfaces are AF_PACKET mode and undefining PERF_TEST is required to see debug logs.

Reply:
If STATIC_ARP with AF_PACKET are always needed to be enabled together then we are missing it in the Dockerfile for non debug build

@ajamshed can you comment on this  requirement? I don't recollect us needing to do this.

Reply:
To make it clear, I had no chance to test the default setup(VFIO device + veth peer). So it may always be needed.

Reply:
Ah right. you did mention you are using AF_PACKET for main interfaces. This mode is not entirely usable. I am not sure if we should spend time on this. @ajamshed

Reply:
@hyunsun,

I just tested `ngic-rtc` in `AF_PACKET` mode (physical ports also bound to kernel driver) on a bare-metal setup. It seems to be running fine. Can you try re-doing the test on bare-metal first? We can probably determine the root cause of the issue more clearly then. In the mean time, I will set up the docker image to check whether it runs fine.

Reply:
@ajamshed Thanks for the confirmation. I've tested multiple times and found the test scenario(ping from UE to SGI network gateway) started working after around 10 failures without STATIC_ARP. I think I can handle this from the test scenario by not counting the first 10~15 ping tries.

BTW, is it supposed to take this long to resolve ARP? Or am I missing some configuration? Here are the logs from DP during the ping failures.
```DP: UL: gtpu packet
DP: Received tunneled packet with teid 0xF0C0FFEE
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: From Ue IP 16.255.255.254
DP: Decap: modified mbuf offset 164, data_len 98, pkt_len98
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[0]:2
DP: Sendto:: ret_arp_data->ip= 13.1.1.254
DP: SGI packet
DP: construct_ether_hdr::arp_queue_unresolved_packet::
	return -1; arp_key.ip= 0xFE01010D
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[0]:2
DP: BEAR_SESS LKUP:DL_KEY ue_addr:13.1.1.254, rid:1
DP: SDF BEAR LKUP FAIL!! DL_KEY ue_addr:13.1.1.254, rid:1
DP: SGI packet
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[0]:2
DP: BEAR_SESS LKUP:DL_KEY ue_addr:13.1.1.1, rid:1
DP: SDF BEAR LKUP FAIL!! DL_KEY ue_addr:13.1.1.1, rid:1```

Reply:
And this is the log when ping starts to work.
```
DP: SGI packet
DP: construct_ether_hdr::arp_queue_unresolved_packet::
	return -1; arp_key.ip= 0xFE01010D
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[0]:2
DP: BEAR_SESS LKUP:DL_KEY ue_addr:13.1.1.254, rid:1
DP: SDF BEAR LKUP FAIL!! DL_KEY ue_addr:13.1.1.254, rid:1
DP: SGI packet
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[0]:2
DP: BEAR_SESS LKUP:DL_KEY ue_addr:13.1.1.1, rid:1
DP: SDF BEAR LKUP FAIL!! DL_KEY ue_addr:13.1.1.1, rid:1
DP: epc_dl.c:epc_dl_set_port_id::
	@SGI:eh->ether_type==ETHER_TYPE_ARP= 0x608
DP: KNI: DL send pkts to kni
DP: epc_ul.c:epc_ul_set_port_id::
	@S1U:eh->ether_type==ETHER_TYPE_ARP= 0x608
DP: KNI: UL send pkts to kni
DP: epc_dl.c:epc_dl_set_port_id::
	@SGI:eh->ether_type==ETHER_TYPE_ARP= 0x608
DP: KNI: DL send pkts to kni
DP: epc_ul.c:epc_ul_set_port_id::
	@S1U:eh->ether_type==ETHER_TYPE_ARP= 0x608
DP: KNI: UL send pkts to kni
DP: epc_dl.c:epc_dl_set_port_id::
	@SGI:eh->ether_type==ETHER_TYPE_ARP= 0x608
DP: epc_dl.c:epc_dl_set_port_id::
	@SGI:eh->ether_type==ETHER_TYPE_ARP= 0x608
DP: KNI: DL send pkts to kni
DP: SGI packet
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[0]:2
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01
DP: SGI packet
DP: SGI packet
DP: SGI packet
DP: SGI packet
DP: SGI packet
DP: SGI packet
DP: SGI packet
DP: SGI packet
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[1]:99999
DP: ACL_LKUP: rid[2]:99999
DP: ACL_LKUP: rid[3]:99999
DP: ACL_LKUP: rid[4]:99999
DP: ACL_LKUP: rid[5]:99999
DP: ACL_LKUP: rid[6]:99999
DP: ACL_LKUP: rid[7]:99999
DP: ACL_LKUP: rid[0]:2
DP: ACL_LKUP: rid[1]:2
DP: ACL_LKUP: rid[2]:2
DP: ACL_LKUP: rid[3]:2
DP: ACL_LKUP: rid[4]:2
DP: ACL_LKUP: rid[5]:2
DP: ACL_LKUP: rid[6]:2
DP: ACL_LKUP: rid[7]:2
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: UL: gtpu packet
DP: Received tunneled packet with teid 0xF0C0FFEE
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: From Ue IP 16.255.255.254
DP: Decap: modified mbuf offset 164, data_len 98, pkt_len98
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[0]:2
DP: MAC found for ip 13.1.1.254, port 1 - 0e:00:00:00:00:01
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01
DP: SGI packet
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[0]:2
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01
DP: UL: gtpu packet
DP: Received tunneled packet with teid 0xF0C0FFEE
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: ASR-SpirentvLS gtpu.c: GPDU_HDR_SIZE 8
DP: From Ue IP 16.255.255.254
DP: Decap: modified mbuf offset 164, data_len 98, pkt_len98
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[0]:2
DP: MAC found for ip 13.1.1.254, port 1 - 0e:00:00:00:00:01
DP: SGI packet
DP: ACL_LKUP: rid[0]:99999
DP: ACL_LKUP: rid[0]:2
DP: BEAR_SESS LKUP:DL_KEY ue_addr:16.255.255.254, rid:1
DP: Encap: modified mbuf offset 92, data_len 134, pkt_len 134
DP: MAC found for ip 119.0.0.253, port 0 - 0a:00:00:00:00:01```